### DISCUSSION TITLE
Possible to use predictable interface names on cloud platforms like VMware?

### DISCUSSION THREAD

**Discussion by jaymzmac** [siderolabs/talos#9744] [Q&A] [answered]
In the [documentation](https://www.talos.dev/v1.8/talos-guides/network/predictable-interface-names/), it mentions that Talos automatically adds `net.ifnames=0` to the kernel arguments for Cloud platforms.

I tried overriding this by setting `net.ifnames=1` in extraKernelArgs, but it appears as though this is ignored and the parameter defined in the source code takes precedence. 

Is my understanding correct, that it is not possible to override `net.ifnames=0` when using a cloud image?

**Comment by smira** [comment, marked as answer]
When using `extraKernelArgs`, you need to do an upgrade (even to the same version) to activate it.

You might need to do `-net.ifnames net.ifnames=1`, or you can adjust in the initial image in the Image Factory using the same syntax.

**Reply by jaymzmac** [reply]
Thanks, `-net.ifnames net.ifnames=1` worked when generating the image in Image Factory.

A follow up question/remark:
I tested Talos v1.8.3 on VMware, and the interface was named "eno16780032" which is not what I'd expect. Interfaces on Linux are typically named "ens192" (depending on the pcie slot assigned to the nic). 

Found this RedHat blog article which explains the root cause and how systemd was patched so that if the nic firmware reports an incorrect index number for onboard devices (like in this case), then an alternative naming scheme is used:
https://www.redhat.com/en/blog/red-hat-enterprise-linux-73-achieving-persistent-and-consistent-network-interface-naming-vmware-environments?extIdCarryOver=true&sc_cid=701f2000001OH7EAAW

Do you think it would be possible to get the same kind of behaviour in Talos as well?

**Reply by smira** [reply]
Talos 1.9 will ship with different (better) `udevd` implementation, so I think there network interface names will become consistent.

**Reply by jaymzmac** [reply]
OK, I'll give 1.9 a try when it's released and report back if it's still not working as expected.
